### DISCUSSION TITLE
Highlighting with different colours in one mol

### DISCUSSION THREAD

**Discussion by dcheung11** [rdkit/rdkit#5740] [General] [answered]
Is it possible to highlight multiple substructures in the same molecule but have each one a different colour? i want to do so using the atom and bond indices 

I tried to do so in react by altering the MoleculeStructure component but i cant get different colours

**Comment by DavidACosgrove** [comment]
It has its own api point - MolDraw2D.DrawMoleculeWithHighlights. IIRC, Greg
did a blogpost about this and other aspects a few years ago called
something similar to “using the new drawing code”. I don’t have access to
my computer to find the exact link but hopefully you can find it with a bit
of googling.
Dave


On Tue, 8 Nov 2022 at 15:46, dcheung11 ***@***.***> wrote:

> Is it possible to highlight multiple substructures in the same molecule
> but have each one a different colour? i want to do so using the atom and
> bond indices
>
> I tried to do so in react by altering the MoleculeStructure component but
> i cant get different colours
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5740>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FSOWGRBNHCUIPQA243WHJYWHANCNFSM6AAAAAAR2O445U>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Reply by dcheung11** [reply]
Thanks Dave- i think i've found it http://rdkit.blogspot.com/2015/02/new-drawing-code.html

Are these examples supported in rdkit.js though? for example, i dont see "highlightAtomColors" on the list of supported options at  https://www.rdkitjs.com/

**Comment by DavidACosgrove** [comment]
Ah, the RDKit.js is a subset of the full toolkit. It normally relies on
people asking for things to be exposed in it, or doing it themselves and
submitting a PR. In the former case, @ptosco normally does the honours, if
and when he gets the time.
Sorry, I had assumed you were working in Python.
Dave


On Tue, 8 Nov 2022 at 19:06, dcheung11 ***@***.***> wrote:

> Thanks Dave- i think i've found it
> http://rdkit.blogspot.com/2015/02/new-drawing-code.html
>
> Are these examples supported in rdkit.js though? for example, i dont see
> "highlightAtomColors" on the list of supported options at
> https://www.rdkitjs.com/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5740#discussioncomment-4090077>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FRXGAFVSJ4ICPG65UDWHKQEBANCNFSM6AAAAAAR2O445U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk

**Comment by ptosco** [comment, marked as answer]
@dcheung11 I have already exposed `highlightAtomColors` to RDKitJS and it should be available in RDKitJS 2022_09_1; see https://github.com/rdkit/rdkit-js/issues/212.

**Reply by dcheung11** [reply]
Oh wow awesome! Does that mean I just need to update the rdkit in my React app?  Or do i need to change something with thte molecularStructure component (from react examples)
Can i just reinstall it as shown in the readMe? (npm i rdkit..), i seem to only be installing   "version": "2022.3.4-1.0.5",. do you know my issue.?

**Reply by MichelML** [reply]
@dcheung11 you need to install rdkit.js 2022.9.1-1.0.0 , and if you use the MoleculeStructure react component from the documentation, you can simply pass highlightAtomColors to the extraDetails prop and this option should work. Hope it helps!

**Reply by dcheung11** [reply]
How do i install a specific version? it seems to default to the 2022.03

**Comment by DavidACosgrove** [comment]
These are two different things, although perhaps it was I who missed the
point. The argument highlightAtomColours for DrawMolecule allows individual
colours to be set for atoms.  The function I mentioned allows for
overlapping colour subsets so that the same atom can have multiple
highlights.


On Wed, 9 Nov 2022 at 03:45, Michel Moreau ***@***.***> wrote:

> @dcheung11 <https://github.com/dcheung11> you need to install rdkit.js
> 2022.9.1-1.0.0 , and if you use the MoleculeStructure component, you can
> simply pass highlightAtomColors to the extraDetails prop and this option
> should work. Hope it helps!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rdkit/rdkit/discussions/5740#discussioncomment-4092885>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACGF2FRIVNALBXNCQACNS3LWHMM3ZANCNFSM6AAAAAAR2O445U>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
David Cosgrove
Freelance computational chemistry and chemoinformatics developer
http://cozchemix.co.uk
